com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Task.java

```diff
@@ -156,29 +164,29 @@ public boolean isCacheable() {
      * A copy of the list of resources will be used and the copy will be sorted
      * before added to the digest. This guarantees that we get the same digest
      * each time given the same set of resources.
-     * @param, digest The digest to update with the resources
+     * @param digest The digest to update with the resources
      * @param resources A list of resources to add
      */
     private void updateDigestWithResources(MessageDigest digest, List<IResource> resources) throws IOException {
         List<IResource> sortedResources = new ArrayList<IResource>(resources);
-        Collections.sort(sortedResources, new Comparator<IResource>() {
+        sortedResources.sort(new Comparator<IResource>() {
             @Override
             public int compare(IResource r1, IResource r2) {
                 return r1.getAbsPath().compareTo(r2.getAbsPath());
             }
         });
         for (IResource r : sortedResources) {
-            digest.update(r.sha1());
+            digest.update(r.sha1(true));
```

Comment:
In such case it makes since to reuse sha1 if it was calculated already (this is what this `true` parameter does). But in some cases we have to recalculate it, so, old behaviour is by default